### ISSUE TITLE
My GSM8K evaluation results different with those reported in the paper

### ISSUE THREAD

**Issue by xiaoxiaostudy** [Yueeeeeeee/HRPO#5]
Hi, thank you for your excellent work!

I’ve been working on reproducing the results reported in the paper using the provided training scripts and settings in the [https://github.com/Yueeeeeeee/HRPO/issues/1#issuecomment-3020884359](url). Below is a summary of my GSM8K evaluation results compared with those reported in the paper:
| Model | Checkpoint | GSM8K Accuracy | Reported (Paper) |
|-------|------------|----------------|------------------|
| Qwen2.5-1.5B-Instruct-gsm8k-group4-lora32-rmin0.99-temp0.5 | 934 | 0.698 | 0.72 |
|  | 750 | 0.694 | 0.72|
| Qwen2.5-1.5B-Instruct-gsm8k-group8-lora32-rmin0.98-temp0.5 | 1750 | 0.660 | 0.72|
| Qwen2.5-1.5B-Instruct-gsm8k-group8-lora32-rmin0.95-temp0.5 | 1000 | 0.667 | 0.72 |
| Qwen2.5-3B-Instruct-gsm8k-group8-lora32-rmin0.95-temp0.5 | 1000 | 0.836 | 0.845 |
| Qwen2.5-3B-Instruct-gsm8k-group8-lora32-rmin0.98-temp0.5 | 1000 | 0.836 | 0.845 |

Are there any additional implementation details or preprocessing steps that could significantly impact performance?

Thanks again for releasing the code — much appreciated!

**Comment by Yueeeeeeee** [comment]
Thank you for your interest in our work. About the GSM8k results, we observe that these can be a bit more sensitive to hyperparameters compared to other datasets. And note there are minor differences between the current release and our original implementation (e.g., we used 0, 42, and 3407 as experiment seeds). I can share the experiment logs and LoRA weights if this could further assist you:)

**Comment by xiaoxiaostudy** [comment]
Thank you for your reply. Have you changed the rank of Lora? I change it to 64 --Qwen2.5-1.5B-Instruct-gsm8k-group4-lora64-rmin0.99-temp0.5 and the ACC in GSM8k is 0.728.

**Comment by Yueeeeeeee** [comment]
Hi, in our intial experiments we tried our lora ranks from 8 to 128 but found that performance gains start to diminish after r=32. As mentioned, GSM8k might be somewhat sensitive to hyperparameters so this might be related.

**Comment by xiaoxiaostudy** [comment]
OK, thank you, your reply is very helpful to me.
